### ISSUE TITLE
Use of ambiguous text on buttons

### ISSUE THREAD

**Issue by rebekahmarkowitz** [WordPress/twentytwentyfour#33]
## Issue Summary

For accessibility reasons, generic buttons on the theme should not have ambiguous text. See the learn more button on this screen shot from the homepage template. 

<img width="651" alt="Screenshot 2023-08-24 at 2 35 08 PM" src="https://github.com/WordPress/twentytwentyfour/assets/93548879/ca2d8e76-156b-4443-8f1b-8601aab41488">. 

We want to guide users to make smart word choices for screen readers where it doesn't make sense/provide clarity out of context.

## Recommendation
Update the button text to match the [WordPress accessibility guidelines for repetitive link text](https://make.wordpress.org/themes/handbook/review/accessibility/required/#repetitive-link-text). 

## Specifications
Template File: Homepage
**WCAG SC:** E.G. [2.4.4](https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-in-context.html) On Input

**Comment by richtabor** [comment]
@rebekahmarkowitz what would you suggest for the button text?

**Comment by joedolson** [comment]
What this should have is contextual text, possibly hidden via screen-reader-text, that provides additional context about the link. This is how most WordPress "read more" links are handled. It's technically only relevant if the pattern appears more than once on a page. The relevant WordPress accessibility-ready guideline and WCAG criteria are both about avoiding links with the same text that link to different targets. 

So this is not a problem in some uses, but could be a problem in others. If an implementer placed this pattern only once on a page, then it's fine. If it is present multiple times with different links, then it's a problem.

Is there a way we can encourage implementers to vary the text at the time of implementation?

**Comment by richtabor** [comment]
> What this should have is contextual text, possibly hidden via screen-reader-text, that provides additional context about the link. This is how most WordPress "read more" links are handled.

This is the button block, not an additional UI/X for pattern insertion. 

I don't think we can enforce buttons to have different text, other than not providing default text at all. It's not likely you'll want two of the same exact patterns, but it's not impossible.

**Comment by thetwopct** [comment]
I've submitted a PR that updates the button text to be descriptive in the context of their use.

**Comment by joedolson** [comment]
I think that what @thetwopct has provided - giving relevant sample text in the pattern - is probably about the best we can do to push authors towards using meaningful link text. I think it would be a good change.

**Comment by richtabor** [comment]
I agree. I like it. Let's keep this issue open though, as more patterns are worked on. Don't want any buttons to slip through.

**Comment by MaggieCabrera** [comment]
All patterns have been pushed and now it's the best time to review the text in them. Please let us know if we can close this or if there are still some buttons that need yet to be edited. I know we've merged a couple PRs that were addressing this but I'm not sure all have been covered

**Comment by luminuu** [comment]
Ping @joedolson @rebekahmarkowitz

**Comment by carolinan** [comment]
The texts are:

- Sign up
- About us
- Subscribe
- Reserve your spot
- Our services


Screenshot of all individual patterns with buttons:
![ collection of patterns with buttons](https://github.com/WordPress/twentytwentyfour/assets/7422055/b7695a98-9445-4314-9b53-12a0f6426a1f)

This pattern is called newsletter landing page and I wasn't sure if this is meant to be an actual landing page or a standalone pattern that can be inserted anywhere
![newsletter block pattern](https://github.com/WordPress/twentytwentyfour/assets/7422055/eb6d5425-f76f-4d4d-b288-72ffcfdb1de9)

**Comment by carolinan** [comment]
Also none of these are functional, they are examples. The user is expected to add the link and when needed, adjust the text.

**Comment by luminuu** [comment]
These ones have been added by the PR of @thetwopct:
- Sign up
- About us
- Our services

I guess these ones are still from the inital commit:
- Subscribe
- Reserve your spot

Is there still work left to do here or are we fine using this button descriptions now?

**Comment by joedolson** [comment]
I think the important characteristic here is that all of the suggested button texts are different and unique to the imagined purpose of the block. That will help encourage users to use unique text. There's nothing we can do to force users to use meaningful link text, so I think this is what we need.
